Library/SharedFunctionsTests.swift

```diff
@@ -182,4 +182,62 @@ final class SharedFunctionsTests: TestCase {
     XCTAssertTrue(ksr_isOSVersionAvailable(12.123))
     XCTAssertTrue(ksr_isOSVersionAvailable(12.9))
   }
+
+  func testAttributedStringWithLinks() {
+    let string = "What a lovely string with a link and another link" as NSString
+    let attrString = NSAttributedString(string: string as String)
+    let links: [AttributedLinkData] = [
+      ("a link", URL(string: "link://a_link"), [.foregroundColor: UIColor.blue]),
+      ("another link", URL(string: "link://another_link"), [:])
+    ]
+    let stringWithLinks = ksr_attributedString(attrString, with: links)
+
+    let aLinkRange = string.range(of: "a link")
+    let anotherLinkRange = string.range(of: "another link")
+    let fullRange = string.range(of: string as String)
+
+    let aLinkAttribute = stringWithLinks
+      .attribute(.link, at: aLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+    let aLinkColorAttribute = stringWithLinks
+      .attribute(.foregroundColor, at: aLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+    let anotherLinkAttribute = stringWithLinks
+      .attribute(.link, at: anotherLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+    let anotherLinkColorAttribute = stringWithLinks
+      .attribute(.foregroundColor, at: anotherLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+
+    XCTAssertNotNil(aLinkAttribute, "a link is a link")
+    XCTAssertNotNil(aLinkColorAttribute, "a link has a foreground color")
```

Comment:
If I'm looking at this right, it looks like this would succeed even if there was not a link but the attributed string had a color. Should we maybe explicitly test the color (values)?